Fix markdown links

marcelklehr opened the pull request:
Ich habe mal einen pull request gemacht, da höchstwarscheinlich noch commits dazu kommen -- es ist ungetestet (TM)

Die Idee ist, einen markdown parser drüberlaufen zu lassen und den parse tree wieder in markdown zu serialisieren, wobei ich im renderer einen filter eingebaut habe, der atrium links in wiki links umbiegt.

marcelklehr committed:
First pass at fixing the markdown links

zonradkuse commented on mdrenderer.py:
```diff
@@ -0,0 +1,204 @@
+    m = self.is_atrium_link.match(link)
+    if m and not image
+      linked_page = self.pages[m.group(1)][0]
+      return self.wiki_link('doc:'+linked_page.build_prefixed_path(), text)
```
Ich habe vorhin gesehen, dass XWiki relative Pfade nimmt wenn man mal eine Seite exportiert. Wenn es zu Problemen kommt, liegt hier vermutlich das Problem.

marcelklehr replied:
Ah, ok. Dann muss ich noch ein pfad diff einbauen...

zonradkuse committed:
Fix import and minor syntactical errors.

marcelklehr commented:
Funktioniert das tatsächlich, bisher? Ich werde heute abend noch den feinschliff machen.

zonradkuse committed:
Fix minor errors.

zonradkuse committed:
Refactor renderer to own variable.

Still getting error that the required positional argument renderer in
__init__ would be missing.

zonradkuse committed:
Add renderer to Lexer constructor.

zonradkuse committed:
Use own init method instead of constructor.

MdRenderer super class constructor does things...

zonradkuse committed:
Add check whether referenced page actually exists.

zonradkuse commented:
Jetzt läuft es auf jeden Fall einmal durch. Inhaltlich habe ich aber noch nichts geprüft. Ich kann dir gleich mal ein Archiv zur Verfügung stellen wenn es durchgelaufen ist. Ich hoffe, das hilft dir so aber schon weiter :)

zonradkuse commented:
Das Archiv liegt in Nextcloud.

zonradkuse commented:
Ich habe gerade mal reingeschaut. Ich glaube, das Ersetzen der Links greift noch nicht. Schau z.B. mal in Archiv OpenAtrium/lg_aachen/Aachen/WebHome.xml

marcelklehr commented on mdrenderer.py:
```diff
@@ -0,0 +1,206 @@
+  def autolink(self, link, is_email=False):
+    m = self.is_atrium_link.match(link)
+    if m and m.group(1) in self.pages:
+      linked_page = self.pages[m.group(1)][0]
```
Warcheinlich ein type problem. Nimmst du als keys für pages_by_node_id strings oder zahlen?

zonradkuse replied:
Ich habe eine type assertion hinzugefügt. Der type sollte int sein. Die type assertion läuft aber nur einmal beim Einlesen der Daten - ich musste das schnell testen^^

zonradkuse committed:
Add type assertion.

marcelklehr committed:
Cast node id to int and calculate relative path

marcelklehr committed:
Fix mind-o

marcelklehr committed:
Three fixes in a row (for ultimate fixery)

 - Relative paths suck (cause there's no parent dir identifier)
 - Paths need to end in an index (i.e. "WebHome")
 - Also grep for ancient pre-subdomain atrium links for maximum joy

marcelklehr committed:
Fix overzealous fixery

(When you think you've fixed it all...)

marcelklehr committed:
Time for space
